src/daft-functions/src/list/bool_and.rs

```diff
@@ -0,0 +1,55 @@
+use common_error::{DaftError, DaftResult};
+use daft_core::{
+    prelude::{Field, Schema},
+    series::Series,
+};
+use daft_dsl::{
+    functions::{ScalarFunction, ScalarUDF},
+    ExprRef,
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub struct ListBoolAnd {}
```

Comment:
Minor nit: we can just declare this as `pub struct ListBoolAnd;` (no trailing `{}`). Just a bit cleaner.